From 2f70d8d796d0018f393654b3296a0eec3be1e98c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:07:27 -0400
Subject: [tools] perf: Get rid of on_exit() feature test

Message-id: <1410181752-25631-203-git-send-email-jolsa@redhat.com>
Patchwork-id: 91836
O-Subject: [PATCH RHEL7.1 BZ1134356 202/307] perf tools: Get rid of on_exit() feature test
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit bac1e4d103f7e3d82e5dc7423c04edcb8c899c22
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon May 12 09:47:25 2014 +0900

description
===========
The on_exit() function was only used in perf record but it's gone in
previous patch.
---
 delete mode 100644 tools/perf/config/feature-checks/test-on-exit.c

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index 2e0d484..e4c85b8 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -30,37 +30,6 @@
 #include <sched.h>
 #include <sys/mman.h>
 
-#ifndef HAVE_ON_EXIT_SUPPORT
-#ifndef ATEXIT_MAX
-#define ATEXIT_MAX 32
-#endif
-static int __on_exit_count = 0;
-typedef void (*on_exit_func_t) (int, void *);
-static on_exit_func_t __on_exit_funcs[ATEXIT_MAX];
-static void *__on_exit_args[ATEXIT_MAX];
-static int __exitcode = 0;
-static void __handle_on_exit_funcs(void);
-static int on_exit(on_exit_func_t function, void *arg);
-#define exit(x) (exit)(__exitcode = (x))
-
-static int on_exit(on_exit_func_t function, void *arg)
-{
- if (__on_exit_count == ATEXIT_MAX)
-  return -ENOMEM;
- else if (__on_exit_count == 0)
-  atexit(__handle_on_exit_funcs);
- __on_exit_funcs[__on_exit_count] = function;
- __on_exit_args[__on_exit_count++] = arg;
- return 0;
-}
-
-static void __handle_on_exit_funcs(void)
-{
- int i;
- for (i = 0; i < __on_exit_count; i++)
-  __on_exit_funcs[i] (__exitcode, __on_exit_args[i]);
-}
-#endif
 
 struct record {
  struct perf_tool tool;
diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index 150c84c..f2edc59 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -174,7 +174,6 @@ CORE_FEATURE_TESTS =   \
  libpython-version  \
  libslang   \
  libunwind   \
- on-exit    \
  stackprotector-all  \
  timerfd    \
  libdw-dwarf-unwind
@@ -200,7 +199,6 @@ VF_FEATURE_TESTS =   \
  libelf-getphdrnum  \
  libelf-mmap   \
  libpython-version  \
- on-exit    \
  stackprotector-all  \
  timerfd    \
  libunwind-debug-frame  \
@@ -571,12 +569,6 @@ ifneq ($(filter -lbfd,$(EXTLIBS)),)
   CFLAGS += -DHAVE_LIBBFD_SUPPORT
 endif
 
-ifndef NO_ON_EXIT
-  ifeq ($(feature-on-exit), 1)
-    CFLAGS += -DHAVE_ON_EXIT_SUPPORT
-  endif
-endif
-
 ifndef NO_BACKTRACE
   ifeq ($(feature-backtrace), 1)
     CFLAGS += -DHAVE_BACKTRACE_SUPPORT
diff --git a/tools/perf/config/feature-checks/Makefile b/tools/perf/config/feature-checks/Makefile
index 2da103c..64c84e5 100644
--- a/tools/perf/config/feature-checks/Makefile
+++ b/tools/perf/config/feature-checks/Makefile
@@ -24,7 +24,6 @@ FILES=     \
  test-libslang.bin  \
  test-libunwind.bin  \
  test-libunwind-debug-frame.bin \
- test-on-exit.bin  \
  test-stackprotector-all.bin \
  test-timerfd.bin  \
  test-libdw-dwarf-unwind.bin
@@ -133,9 +132,6 @@ test-liberty-z.bin:
 test-cplus-demangle.bin:
  $(BUILD) -liberty
 
-test-on-exit.bin:
- $(BUILD)
-
 test-backtrace.bin:
  $(BUILD)
 
diff --git a/tools/perf/config/feature-checks/test-all.c b/tools/perf/config/feature-checks/test-all.c
index c28d99d..086001b 100644
--- a/tools/perf/config/feature-checks/test-all.c
+++ b/tools/perf/config/feature-checks/test-all.c
@@ -73,10 +73,6 @@
 # include "test-libbfd.c"
 #undef main
 
-#define main main_test_on_exit
-# include "test-on-exit.c"
-#undef main
-
 #define main main_test_backtrace
 # include "test-backtrace.c"
 #undef main
@@ -114,7 +110,6 @@ int main(int argc, char *argv[])
  main_test_gtk2(argc, argv);
  main_test_gtk2_infobar(argc, argv);
  main_test_libbfd();
- main_test_on_exit();
  main_test_backtrace();
  main_test_libnuma();
  main_test_timerfd();
diff --git a/tools/perf/config/feature-checks/test-on-exit.c b/tools/perf/config/feature-checks/test-on-exit.c
deleted file mode 100644
index 8e88b16..0000000
--- a/tools/perf/config/feature-checks/test-on-exit.c
+++ /dev/null
@@ -1,16 +0,0 @@
-#include <stdio.h>
-#include <stdlib.h>
-
-static void exit_fn(int status, void *__data)
-{
- printf("exit status: %d, data: %d\n", status, *(int *)__data);
-}
-
-static int data = 123;
-
-int main(void)
-{
- on_exit(exit_fn, &data);
-
- return 321;
-}
-- 
1.7.1